druid-shell/src/platform/windows/window.rs

```diff
@@ -192,6 +192,20 @@ struct DCompState {
 /// Message indicating there are idle tasks to run.
 const XI_RUN_IDLE: UINT = WM_USER;
 
+/// Message indicating explit destroy windows
```

Comment:
The change is good but I'd like to rewrite the comment to explain more clearly what's happening. I think the issue is related to the stackoverflow thread, but different - we're not relying on the message being available in `PeekMessage`, this is really more about reentrancy.

I wanted to make this as a [suggestion](https://github.com/isaacs/github/issues/1452) but apparently that only supports a single line.

```
/// Message relaying a request to destroy the window
///
/// Calling `DestroyWindow` from inside the handler is problematic
/// because it will recursively cause a `WM_DESTROY` message to be
/// sent to the window procedure, even while the handler is borrowed.
/// Thus, the message is dropped and the handler doesn't run.
///
/// As a solution, instead of immediately calling `DestroyWindow`, we
/// send this message to request destroying the window, so that at the
/// time it is handled, we can successfully borrow the handler.
```

Reply:
Fixed :)